app/src/main/java/com/sdp13epfl2021/projmag/activities/PreferencesActivity.kt

```diff
@@ -0,0 +1,118 @@
+package com.sdp13epfl2021.projmag.activities
+
+import android.os.Bundle
+import android.util.Log
+import android.view.View
+import android.widget.Button
+import android.widget.CheckBox
+import android.widget.TextView
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import com.sdp13epfl2021.projmag.R
+import com.sdp13epfl2021.projmag.database.Utils
+import com.sdp13epfl2021.projmag.model.ProjectFilter
+
+/**
+ * An activity where the user can set it's preferences for projects.
+ * This preferences are pushed to the Datatbase
+ */
+class PreferencesActivity : AppCompatActivity() {
+    /**
+     * The database of user's informations
+     */
+    private val userDB = Utils.getInstance(this).userDataDatabase
+
+    /**
+     * Check box that require the project to ask for a bachelor degree
+     */
+    private lateinit var bachelor: CheckBox
+
+    /**
+     * Check box that require the project to ask for a master degree
+     */
+    private lateinit var master: CheckBox
+
+    /**
+     * Check box that require the project to be applied by the user
+     */
+    private lateinit var applied: CheckBox
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.project_filter_settings)
+
+        // Update the view for this activity
+        findViewById<TextView>(R.id.filter_title).text = getString(R.string.preferences)
+        findViewById<View>(R.id.filter_preferences_switch).visibility = View.GONE
```

Comment:
When I test on the emulator I think that the expected behavior is inverted when I activate preferences the preference options disappear and when I deactivate it they appear